update readme for unified configuration (#23452)
Azure/azure-sdk-for-java: sdk/spring/azure-spring-boot-starter-keyvault-secrets/README.md

Key Vault also supports a contentType field for secrets. Clients may specify the content type of a secret to assist in interpreting the secret data when it's retrieved. The maximum length of this field is 255 characters. There are no pre-defined values. The suggested usage is as a hint for interpreting the secret data.

Besides, this starter provides features of supporting multiple Key Vaults, case sensitive mode of Key Vault names and using placeholder presenting Key Vault names in property file

### Configuration Options

> Azure Spring Boot Key Vault Starter deprecates all legacy properties of which the prefix is `azure.keyvault` and uses `spring.cloud.azure.keyvault` instead.

Reviewer: tagging @vcolin7 for more feedback

Reviewer 2: We should keep the name consistent across the README.
```suggestion
Azure Key Vault Secrets Spring Boot Starter deprecates all legacy properties of which the prefix is `azure.keyvault` and uses `spring.cloud.azure.keyvault` instead.
```

> When a deprecated property is detected while its active property is not found, the active property will be configured into application environment with value from the deprecated property.

Reviewer 2: ```suggestion
When a deprecated property is detected while its active property is not found, the active property will be configured into the application environment with the value from the deprecated property.
```

> If you load configuration properties from Azure Key Vault, the preceding detection and replacement are also applicable for Key Vault property sources. Replaced properties from Key Vault have higher priorities than local ones.

Reviewer 2: ```suggestion
If you load configuration properties from Azure Key Vault, the preceding detection and replacement are also applicable for Key Vault property sources. Replaced properties from Key Vault have a higher priority than local ones.
```

spring.cloud.azure.keyvault.credential.tenant-id|Tenant id for the Azure resources.|||
spring.cloud.azure.keyvault.enabled|To turn on/off Azure Key Vault Secret property source.|true||
spring.cloud.azure.keyvault.environment.authority-host|Authority Host URI|https://login.microsoftonline.com/||
spring.cloud.azure.keyvault.order|Define the order of the key vaults you are delivering (comma delimited, e.g 'my-vault, my-vault-2').|||
spring.cloud.azure.keyvault.refresh-interval|Interval for PropertySource to refresh secret keys|1800000(ms)||
spring.cloud.azure.keyvault.secret-keys|If application using specific secret keys. If this property is set, application will only load the keys in the property and won't load all the keys from keyvault, that means if you want to update your secrets, you need to restart the application rather than only add secrets in the keyvault.|||
spring.cloud.azure.keyvault.secret-service-version|Valid secret-service-version value can be found [here][version_link].|The latest value||
spring.cloud.azure.keyvault.uri|Azure Key Vault Uri.|||

> Note: for multiple Key Vault usage, specify your Key Vault name after the prefix of `spring.cloud.azure.keyvault`.

Reviewer 2: It would be nice to have an example of what you mean here, otherwise it's not clear if people should do something like `spring.cloud.azure.keyvault.mykeyvault` or `spring.cloud.azure.keyvaultmykeyvault`.

#### Deprecated Properties

> |Name|Description|Default Value|Comment|

Reviewer: Consider renaming Name to Obsolete Type and Comment to Replacement Type

spring.cloud.azure.keyvault.environment.authority-host|Authority Host URI|https://login.microsoftonline.com/|
spring.cloud.azure.keyvault.order|Define the order of the key vaults you are delivering (comma delimited, e.g 'my-vault, my-vault-2').||
spring.cloud.azure.keyvault.refresh-interval|Interval for PropertySource to refresh secret keys|1800000(ms)|
spring.cloud.azure.keyvault.secret-keys|If application using specific secret keys. If this property is set, application will only load the keys in the property and won't load all the keys from keyvault, that means if you want to update your secrets, you need to restart the application rather than only add secrets in the keyvault.||
spring.cloud.azure.keyvault.secret-service-version|Valid secret-service-version value can be found [here][version_link].|The latest value|
spring.cloud.azure.keyvault.uri|Azure Key Vault Uri.||

Note: for multiple Key Vault usage, specify your Key Vault name after the prefix of `spring.cloud.azure.keyvault.`. For usage examples, please refer to section [Multiple Key Vault support](#multiple-key-vault-support-1).
#### Deprecated Properties

> |Obsolete Type|Description|Default Value|Obsolete Type|

Reviewer 3: The second `Obsolete Type` should be `replacement`?

azure.keyvault.enabled|To turn on/off Azure Key Vault Secret property source.|true|**spring.cloud.azure.keyvault.enabled** |
azure.keyvault.order|Define the order of the key vaults you are delivering (comma delimited, e.g 'my-vault, my-vault-2').| |**spring.cloud.azure.keyvault.order** |
azure.keyvault.refresh-interval|Interval for PropertySource to refresh secret keys|1800000(ms) |**spring.cloud.azure.keyvault.refresh-interval** |
azure.keyvault.secret-keys|If application using specific secret keys. If this property is set, application will only load the keys in the property and won't load all the keys from keyvault, that means if you want to update your secrets, you need to restart the application rather than only add secrets in the keyvault.| |**spring.cloud.azure.keyvault.secret-keys** |
azure.keyvault.secret-service-version|Valid secret-service-version value can be found [here][version_link].|The latest value |**spring.cloud.azure.keyvault.secret-service-version** |
azure.keyvault.tenant-id|Tenant id for the Azure resources.| |**spring.cloud.azure.keyvault.credential.tenant-id** |

> azure.keyvault.uri|Azure Key Vault Uri.| |**spring.cloud.azure.keyvault.uri** |

Reviewer 3: URI and Uri, maybe we should make them consistent.

Author: Updated for URI now.
